editor/src/clj/editor/gui.clj

```diff
@@ -567,32 +598,86 @@
    (or (prop-resource-error node-id prop-kw prop-value prop-name)
        (validation/prop-error :fatal node-id prop-kw validation/prop-resource-ext? prop-value resource-ext prop-name))))
 
-;; SDK api
-(defn references-gui-resource? [evaluation-context node-id prop-kw gui-resource-name]
-  (and (g/property-value-origin? (:basis evaluation-context) node-id prop-kw)
-       (= gui-resource-name (g/node-value node-id prop-kw evaluation-context))))
-
```

Comment:
I've changed how this works. The new system supports updating references to renamed gui resources inside template overrides and layouts, and even across scenes as long as the gui resource name is not shadowed by another resource in a referencing gui.

I added a bunch of tests for this in `integration.gui-test/template-layout-resource-rename-test`.